src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2638,6 +2638,63 @@ public static void UsingAlternativeTimeZoneIdsTest(string windowsId, string iana
             }
         }
 
+        public static bool SupportIanaNamesConversion => PlatformDetection.IsNotBrowser && PlatformDetection.ICUVersion.Major >= 52;
+
+        [ConditionalFact(nameof(SupportIanaNamesConversion))]
+        public static void IsIanaIdTest()
+        {
+            bool expected = !s_isWindows;
+
+            foreach (TimeZoneInfo tzi in TimeZoneInfo.GetSystemTimeZones())
+            {
+                Assert.True((expected || tzi.Id.Equals("Utc", StringComparison.OrdinalIgnoreCase)) == tzi.HasIanaId, $"`{tzi.Id}` has wrong IANA Id indicator");
+            }
+
+            Assert.False(TimeZoneInfo.FindSystemTimeZoneById("Pacific Standard Time").HasIanaId, $"`Pacific Standard Time` should not be IANA Id.");
+            Assert.True(TimeZoneInfo.FindSystemTimeZoneById("America/Los_Angeles").HasIanaId, $"'America/Los_Angeles' should be IANA Id");
+        }
+
+        [ConditionalTheory(nameof(SupportIanaNamesConversion))]
+        [InlineData("Pacific Standard Time", "America/Los_Angeles")]
```

Comment:
Is this the same data that is used in `UsingAlternativeTimeZoneIdsTest` above? Can we refactor it and combine it so it is only defined once?

Reply:
It has some difference. this one I need it to round trip while the other doesn't have to round trip. I am expecting in the future we may need to add more and both data will diverge. So, I prefer to keep it liek that.